src/ioc/dbtemplate/msi.cpp

```diff
@@ -300,7 +300,7 @@ static void makeSubstitutions(inputData * const inputPvt,
             char *pstart;
             char *pend;
             int  cmdind=-1;
-            int  i;
+            long unsigned int  i;
```

Comment:
Don't you now get a warning on line 307 with `cmdind = i;` instead? It might be you need to leave it as integer and put an integer cast on NELEMENTS.

Reply:
Actually no.

Do you think is better to use the cast ?

Reply:
I get a warning on windows if I make the change
```
C:\devel\epics-base\src\ioc\dbtemplate\msi.cpp(307): warning C4365: '=': conversion from 'unsigned long' to 'int', signe
d/unsigned mismatch
```
given `cmdind` needs to be signed for `-1` there will need to be a cast somewhere. Maybe using `size_t i` and then putting the cast on `cmdline = (int)i` is better than on `NELEMENTS`

Reply:
Looks good. I've done that.